### DISCUSSION TITLE
getFrameKinematicHessian

### DISCUSSION THREAD

**Discussion by Cfather** [stack-of-tasks/pinocchio#2489] [Q&A] [answered]
Hi, 

I wonder if there is a `getFrameKinematicHessian` that computes the hessian tensor of a frame kinematics (an additional SE3 placement with respect to a specific joint), that is similar to the existing function `getJointKinematicHessian`?
I found functions like `getFrameJacobianTimeDerivative` or `getFrameAcceleration` but I'm not sure if this is the same as what I want here.

Thanks!

**Comment by jcarpent** [comment, marked as answer]
> I wonder if there is a `getFrameKinematicHessian` that computes the hessian tensor of a frame kinematics (an additional SE3 placement with respect to a specific joint), that is similar to the existing function `getJointKinematicHessian`?

Yes, it is the same algorithm behind, except that Frame takes into account a relative placement wrt a joint frame.

**Reply by Cfather** [reply]
Thanks for your reply! Is there an existing implementation around already in pinocchio? Or it's not that hard to implement if we just consider the relative placement? I can try submit a pull request about this if it's not there in pinocchio yet.

**Comment by jcarpent** [comment]
> I found functions like `getFrameJacobianTimeDerivative` or `getFrameAcceleration` but I'm not sure if this is the same as what I want here.

Have you looked to the document of these two functions?

**Reply by Cfather** [reply]
Yes, I have. But I would say whatI want is the hessian of the frame kinematics, which should be a 6 x model.nv x model.nv tensor. 
These two functions compute the hessian implicitly. As a result, they are not directly what I want.
